cdr/cdr_pgsql.c

```diff
@@ -783,12 +783,19 @@ static int config_module(int reload)
 
 static int load_module(void)
 {
-	ast_cli_register_multiple(cdr_pgsql_status_cli, sizeof(cdr_pgsql_status_cli) / sizeof(struct ast_cli_entry));
+    int res = 0;
+
 	if (config_module(0)) {
-		return AST_MODULE_LOAD_DECLINE;
+		res = AST_MODULE_LOAD_DECLINE;
+	} else {
+		if(ast_cdr_register(name, ast_module_info->description, pgsql_log)) {
+			res = AST_MODULE_LOAD_DECLINE;
+		} else {
+			ast_cli_register_multiple(cdr_pgsql_status_cli, sizeof(cdr_pgsql_status_cli) / sizeof(struct ast_cli_entry));
+		}
 	}
-	return ast_cdr_register(name, ast_module_info->description, pgsql_log)
-		? AST_MODULE_LOAD_DECLINE : 0;
+
+    return res;
```

Comment:
If you are returning anything other than `AST_MODULE_LOAD_SUCCESS` you need to call `unload_module()` before returning, otherwise we leak memory that is allocated by `config_module(...)`.

Reply:
Done, thanks!